### PR TITLE
website: Migration guide updates for provider.DataSourceType and provider.ResourceType deprecation

### PR DESCRIPTION
**Pull request by bflad**
Reference: https://github.com/hashicorp/terraform-plugin-framework/pull/472

Note: The source code reference links to the http, random, and tls providers will be out of date until they are migrated after this functionality is released in the framework. I'm guessing we should hold off merging this until after they are migrated?

### CODE REVIEW

**Comment by bendbennett** [comment]
> Reference: #472
> 
> Note: The source code reference links to the http, random, and tls providers will be out of date until they are migrated after this functionality is released in the framework. I'm guessing we should hold off merging this until after they are migrated?

We could update the migration docs to remove source code links and replace with generic code examples?

File: website/docs/plugin/framework/migrating/data-sources/index.mdx [hunk 1]
```diff
@@ -45,48 +45,49 @@ schema.Resource {
...
 
-The `GetSchema` function returns a `tfsdk.Schema` struct which defines your data source's attributes. This is the same
+The `GetSchema` method returns a `tfsdk.Schema` struct which defines your data source's attributes. This is the same
 struct you use to define provider and resource attributes.
 
-The `NewResource` function returns a type that you define. The type implements the `resource.Resource` interface,
-including the CRUD functions for your resource.
+The `TypeName` method returns a type name that you define.
+
+For `Read` method implements the logic for writing into the Terraform state.
```
Comment (bendbennett) [inline website/docs/plugin/framework/migrating/data-sources/index.mdx:71]:
```suggestion
The `Read` method implements the logic for writing into the Terraform state.
```

**Review by bendbennett — APPROVED** [review]
LGTM

**Comment by bflad** [comment]
I think I caught all the links pointing to code that was updated.

> We could update the migration docs to remove source code links and replace with generic code examples?

I'm generally a proponent of this approach as we can show code with only the necessary context and without needing to update links while the framework is under development, but that can be an exercise done separately. 👍

**Pull request merged**